packages/integration-test/src/setup.js

```diff
@@ -0,0 +1,174 @@
+/* eslint-disable func-names */
+/* global web3 */
+const { getContractAddressesForNetwork, NetworkId: networkIDs } = require('@aztec/contract-addresses');
+const contractArtifacts = require('@aztec/contract-artifacts');
+const {
+    errors: { codes, AztecError },
+} = require('@aztec/dev-utils');
+const TruffleContract = require('@truffle/contract');
+
+const { capitaliseFirstChar } = require('./utils');
+
+class Setup {
+    /**
+     * Sets up the environment for an integration test. Specifically, it creates Truffle Contract instances
+     * from the aztec contract-artifacts; at the appropriate address in the contract-addresses package.
+     *
+     * Can be used to setup a local development testing environment for an integration test
+     *
+     * @param {Object} config - config used to modify the test environment setup
+     * @param {String} accounts - Ethereum addresses to be used in the test
+     */
+    constructor(config, accounts) {
```

Comment:
The config situation can be simplified, by having the constructor take in an optional object with the config values, and otherwise having defaults as state in the class. Would fit the idea of having data + related logic in one place.

Reply:
Yep, this is a really good idea. Have implemented